src/plugins/embeddable/public/lib/actions/edit_panel_action.ts

```diff
@@ -35,11 +35,18 @@ export class EditPanelAction implements Action<ActionContext> {
   public readonly type = ACTION_EDIT_PANEL;
   public readonly id = ACTION_EDIT_PANEL;
   public order = 50;
+  public currentAppId: string | undefined;
 
   constructor(
     private readonly getEmbeddableFactory: EmbeddableStart['getEmbeddableFactory'],
     private readonly application: ApplicationStart
-  ) {}
+  ) {
+    if (this.application?.currentAppId$) {
+      this.application.currentAppId$.subscribe(
```

Comment:
I think we have to somehow unsubscribe here, otherwise will get a memory leak
maybe we could just `.pipe(first())` to make sure it unsubscribed itself (if that is sufficient) 
same here: https://github.com/elastic/kibana/pull/62865/files#diff-85e9be7a6b60df87e48500d65f596d1aR101

Reply:
Oh, good eye. The subscription in `edit_panel_action` I've cleaned up by using a pipe as suggested.

The subscription in `visualize_embeddable_factory` is a little different, as it needs to update correctly when the app id changes. To unsubscribe that, I have a method in the factory called `unsubscribeSubscriptions`. The plugin keeps a reference to the factory, then calls that method in its `stop` method. This was the best way I could think of to clean up that subscription.

Reply:
The `this.application.currentAppId$` is actually a `BehaviorSubject` it always has a value and it can be accessed synchronously.

My guess is that you should be able to do

```ts
let originatingAppParam = getObservableCurrentValue(this.params.start().core.application.currentAppId$);
```

where

```ts
const getObservableCurrentValue = (obs$) => {
  let value;
  obs$.pipe(take(1)).subscribe(anotherValue => value = anotherValue);
  return value;
};
```